inc/azure_c_util/interlocked.h

```diff
@@ -29,7 +30,7 @@ MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, de
 MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_high, int64_t, exchange_low, int64_t*, comperand_result);
```

Comment:
should be protetced just like its implementation, otherwise it is only detectable at link time,

missing #ifdef _WIN64